components/BarcodeScanner.vue

```diff
@@ -138,28 +155,46 @@ export default {
       cameraConstraints.zoom.step = step;
     }
 
+    async function emitScanResult({barcodeDetector, video}) {
+      const {signal} = abortController;
+      const barcodes = await detectBarcodes({barcodeDetector, video, signal});
+      const [result] = barcodes;
+      emit('result', {type: result.format, text: result.rawValue});
+    }
+
+    function startBarcodeDetection() {
+      // Get video element
+      const video = document.querySelector(
+        '#dce-video-container > video');
+      const {BarcodeDetector} = globalThis;
+      // Check if BarcodeDetector is supported
+      if(!BarcodeDetector) {
+        alert('Barcode Detector is not supported in this browser.');
+        return;
+      }
+      const barcodeDetector = new BarcodeDetector({
+        formats: formatsToSupport
+      });
+      // emit the first scanned result
+      emitScanResult({barcodeDetector, video});
+    }
+
     // Update camera zoom
     async function onZoomChange(updatedValue) {
       scanner.applyVideoConstraints({
         advanced: [{zoom: updatedValue}],
       });
     }
 
-    function onScanSuccess(decodedText, decodedResult) {
-      const text = decodedText;
-      const type = decodedResult?.result?.format?.formatName;
-      console.log(
-        'BarcodeScanner detected something:',
-        decodedText ? decodedText : '<empty string>'
-      );
-      if(!type || !text) {
-        return;
-      }
-      emit('result', {type, text});
+    // Use Barcode Detection API instead of html5qrcode's logic
```

Comment:
Is this comment out of place? If not, style nit ... otherwise we can either move it elsewhere or just remove it.

```suggestion
    // use Barcode Detection API instead of html5qrcode's logic
```